lib/index.js

```diff
@@ -53,8 +53,10 @@ class TeamcityReporter {
             let body;
             try { body = JSON.parse(this.currItem.response.body) } catch (e) { body = '' };
             console.log('Response body:', body);
+            console.log(`##teamcity[testFinished name='${args.item.name}' duration='${this.currItem.response.responseTime}']`);
+        } else {
+            console.log(`##teamcity[testFailed name='${args.item.name}' errorDetails='${err}' status='ERROR']`);
         }
-        console.log(`##teamcity[testFinished name='${args.item.name}' duration='${this.currItem.response.responseTime}']`);
```

Comment:
This doesn't look quite right to me because the testFinished message needs to be logged regardless of whether the test passes or not. After this change we will not log testFinished if the test does not pass. I've added a null-response branch that has a simpler change that I think solves the core problem.